Baikal/PostEffects/ML/denoiser.cpp

```diff
@@ -63,30 +60,30 @@ namespace Baikal
             }
 
             return std::make_unique<InferenceImpl>(model_path,
-                                                   gpu_memory_fraction,
-                                                   visible_devices,
-                                                   width,
-                                                   height,
+                                                   params.gpu_memory_fraction,
+                                                   params.visible_devices,
+                                                   params.width,
+                                                   params.height,
                                                    input_channels);
         }
 
-        MLDenoiser::MLDenoiser(const CLWContext& context, Inference::Ptr inference, MLDenoiserInputs inputs)
-                   : m_inference(std::move(inference))
+        MLDenoiser::MLDenoiser(CLWContext context, const MLDenoiserParams& params)
```

Comment:
Passing by value here kills the idea of forward declaration. It was a comment on Alexander's code, and now it is back.